src/resolve/resolved-conf.c

```diff
@@ -414,3 +414,46 @@ int manager_parse_config_file(Manager *m) {
         return 0;
 
 }
+
+int config_parse_refuse_record_types(
+                const char *unit,
+                const char *filename,
+                unsigned line,
+                const char *section,
+                unsigned section_line,
+                const char *lvalue,
+                int ltype,
+                const char *rvalue,
+                void *data,
+                void *userdata) {
+
+        Manager *m = ASSERT_PTR(userdata);
+        int r;
+        Set *refused_records = NULL;
+
+        refused_records = set_free(refused_records);
+
+        for (const char *p = rvalue;;) {
+               _cleanup_free_ char *word = NULL;
+               r = extract_first_word(&p, &word, ",", EXTRACT_UNQUOTE);
+               if (r < 0)
+                       return log_syntax_parse_error(unit, filename, line, r, lvalue, rvalue);
+
+               if (r == 0)
+                       break;
+
+               r = dns_type_from_string(word);
+               if (r < 0) {
+                     log_syntax(unit, LOG_WARNING, filename, line, r, "Invalid DNS record type, ignoring: %s", word);
+                     continue;
+               }
+
+               r = set_ensure_put(&refused_records, NULL, INT_TO_PTR(r));
```

Comment:
honestly, just add what you parsed here directly to `m->refuse_record_types`.

I mean, if you do:

```
RefuseRecordTypes=AAAA MX
RefuseRecordTypes=A
```

then this should be equivalent to:

```
RefuseRecordTypes=A AAAA MX
```

And in your current code it would be, the 2nd line would completely override the first, but is not merged like it should (and how this is usually done in our config files)